src/vm/memory/memory.zig

```diff
@@ -108,19 +120,19 @@ pub const Memory = struct {
         MemoryOutOfBounds,
     }!void {
 
-        // Check if the address is valid.
+        // Insert the value into the memory.
         if (address.segment_index < 0) {
-            return CairoVMError.InvalidMemoryAddress;
+            self.temp_data.put(address, value) catch {
+                return CairoVMError.MemoryOutOfBounds;
```

Comment:
Do we need the `InvalidMemoryAddress` error or we put these under the same umbrella of `MemoryOutOfBounds`?

Reply:
If the error isn't used anymore remove it from `vm/error.zig`

Reply:
I saw this earlier but I said to myself that we would put it in https://github.com/keep-starknet-strange/cairo-zig/issues/124 because the `set` function anyway here is not complete and therefore cannot be tested properly.

Indeed, here we always catch the put error in a `MemoryOutOfBounds` but I think that we must first add all the tests and all the verifications properly as in the Python or Rust implementation before being able to rule on this problem which is not as simple as that in my opinion because before `put` we must first do a certain number of checks with the potential return of errors.

Reply:
OK fair enough, I'll approve then, let's review the used and unused errors at the end of first milestone